pkg/kubelet/pod_manager.go

```diff
@@ -110,7 +126,37 @@ func (self *basicPodManager) UpdatePods(u PodUpdate, podSyncTypes map[types.UID]
 	}
 }
 
-func updatePods(changed []api.Pod, current []api.Pod) []api.Pod {
+// Need to acquire the lock before calling this function.
+func (self *basicPodManager) SetPods(newPods []api.Pod) {
+	pods := []api.Pod{}
```

Comment:
Actually, when the slice gets re-allocated the pointer in the map is no longer a pointer into the slice we have. So any changes will not be reflected:

http://play.golang.org/p/JWCGIQq_ge